modules/command-engine/scaladsl/src/main/scala/surge/scaladsl/event/SurgeEvent.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package surge.scaladsl.event
+
+import akka.actor.ActorSystem
+import com.typesafe.config.{ Config, ConfigFactory }
+import surge.core
+import surge.core.event.{ SurgeEventServiceImpl, SurgeEventServiceModel }
+import surge.metrics.Metric
+import surge.scaladsl.common.HealthCheckTrait
+
+import scala.jdk.CollectionConverters._
+
+trait SurgeEvent[AggId, Agg, Evt] extends core.SurgeProcessingTrait[Agg, Nothing, Nothing, Evt] with HealthCheckTrait {
+  def aggregateFor(aggregateId: AggId): AggregateRef[Agg, Evt]
+  def getMetrics: java.util.List[Metric]
```

Comment:
Since this is the scaladsl we can expose a scala `List` here

Reply:
Are you OK with `immutable.Vector`?  Under the covers it converts a map `Iterator` to a `Vector`. There probably isn't much to be gained by converting it to `List` here.

Reply:
Yep - I'm good with a `Vector` here for the type